src/modules/previewpane/powerpreview/powerpreview.cpp

```diff
@@ -75,28 +75,43 @@ void PowerPreviewModule::set_config(const wchar_t* config)
     }
 }
 
-// Enable the powertoy
+// Enable preview handlers.
 void PowerPreviewModule::enable()
```

Comment:
Should PowerPreviewModule::enable() enable all the preview handlers? I thought it would behave how the is_enabled function is behaving, i.e. it will enable whichever preview handlers were enabled within the PowerPreview settings. (Example if I have only enabled .md preview in the powerpreview settings, then enabling the option in the general settings should only enable that.)

Reply:
The enable in this project works differently, if the .md is already enabled, there would not be any need to enable again, do you agree? The definition in the case we are only managing one powertoy, but in this case we are managing multiple PowerToys. Should we enable .md when it is already enabled? .md can be enabled separately under the these settings:

![image](https://user-images.githubusercontent.com/58791731/74986443-397c7100-53ee-11ea-98c3-a59d5f95a792.png)

Reply:
I would say it should behave similar to how we do the same for other PowerToys. For example when the PowerToys.exe is closed everything is disabled. But every time you relaunch the PowerToys.exe runner the saved settings are used instead of re-enabling all the power-toys. Similar by turning off FileExplorerPreview from the general settings it should disable all, but re-enabling it should enable all those that was enabled before the last disable.
For example:
I initially have only Show SVG on and Show Markdown is off. Then I turn of file explorer preview from the general settings temporarily due to some reason. This should ideally not affect the view in the screenshot you sent since those are specific settings. So if I re-enable File Explorer Preview from general settings, the File Explorer Preview Settings should remain unchanged (i.e.Show SVG is still on but Show MD is off).
Essentially my idea is that to check if SVG preview panes should be shown, both the General settings File Explorer Preview should be enabled and the Show SVG should be enabled. If either is false then it should be disabled.

Let me know if you agree with this? We could also get a PM perspective on this for UX point of view.

Reply:
Oh i see thanks.

Reply:
Updated here: https://github.com/microsoft/PowerToys/pull/1319/commits/fde818e4e993a1b18191df0d113a3e869f62d60b

Enable now loads the saved settings json file by calling the init_settings() method.